typing/types.mli

```diff
@@ -331,6 +333,12 @@ and constructor_arguments =
   | Cstr_tuple of type_expr list
   | Cstr_record of label_declaration list
 
+and unboxed_status =
```

Comment:
I don't think there would be any significant memory overhead with a record since the 4 possible values will be stored as structured constants (and thus shared).  Are you concerned with the extra memory load?

And now that the worm is in the fruit, I expect some `@@packed` attribute so that record fields that can be represented with a few bits can be merged in a single word (and then the record `@unboxed` if a single word remain after packing) :smile: 

But seriously, it's fine to keep the 4 constructors.